web/modules/custom/joinup_search/joinup_search.module

```diff
@@ -132,13 +132,17 @@ function _joinup_search_search_api_has_condition($conditions, $field, $value, $o
  * boolean and "s" for single value, as they are used for sorting.
  */
 function joinup_search_search_api_solr_documents_alter(&$documents, IndexInterface $index, array $items) {
+  /** @var \Drupal\joinup_front_page\FrontPageMenuHelperInterface $front_page_helper */
+  $front_page_helper = \Drupal::service('joinup_front_page.front_page_helper');
   /** @var \Solarium\QueryType\Update\Query\Document\Document[] $documents */
   foreach ($documents as $document) {
     if (empty($document->getFields()['bs_sticky'])) {
       $document->setField('bs_sticky', FALSE);
     }
-    if (!isset($document->getFields()['bs_site_pinned'])) {
-      $document->setField('bs_site_pinned', FALSE);
+    if (!empty($document->getFields()['ss_search_api_id'])) {
+      /** @var \Drupal\search_api\Item\Item[] $items */
+      $id = $document->getFields()['ss_search_api_id'];
+      $document->setField('bs_site_pinned', (bool) $front_page_helper->getFrontPageMenuItem($items[$id]->getOriginalObject()->getValue()));
```

Comment:
I'm sad that we are still patching and altering rather than providing robust APIs. Providing a computed field would had been a far better solution as it would provide a reusable functionality, i.e. getting the pin weight by simply invoking `$entity->pinned` which would return the menu item weight or `NULL`, if not pinned. At least this was my understanding, after our agreement.

Reply:
The PR is not ready yet. I found this line which makes it too easy to patch it quickly so I made one change to quickly have a green test before moving to the computed field as we agreed.

Reply:
Keep in mind that though a computed field has been created, this cannot be automated as either the computed field has to be applied to all indexable entities, or this solution has to be applied (because all other entities have to retrieve the value 0).